src/middlewared/middlewared/plugins/network.py

```diff
@@ -1706,8 +1706,49 @@ def choices(self, options):
         except Exception:
             self.logger.info('Failed to sync routes', exc_info=True)
 
+        if not await self.middleware.call('system.is_freenas') and await self.middleware.call('failover.licensed'):
+            await self.disable_evil_nic_capabilities()
+
         await self.middleware.call_hook('interface.post_sync')
 
+    @private
+    async def disable_evil_nic_capabilities(self):
+        """
+        When certain NIC's are added to a bridge or other members are added to a bridge when these NIC's are already
+        on the bridge, bridge brings all interfaces into lowest common denominator which results in a network hiccup.
+        This hiccup in case of failover makes backup node come ONLINE as master as there's a hiccup in the
+        master/backup communication. This scenario is common to vnet/nat based jails and VM's, this method checks
+        if the user has such VM's or jails which can bring forward this case and disables certain capabilities for
+        the affected NIC's so that the user is not affected by the interruption which is caused when these NIC's
+        experience a hiccup in the network traffic.
+        """
+        nics = await self.middleware.call(
+            'jail.nic_capability_checks',
+            None if await self.middleware.call('failover.status') == 'MASTER' else await self.middleware.call(
+                'failover.call_remote', 'jail.query', [['OR', [['vnet', '=', 1], ['nat', '=', 1]]]]
```

Comment:
This seems a bit redundant, regarding the call and filters. Should we move all of this inside jail.nic_capability_checks ?

Reply:
Good idea, I'll update the PR